monitor/service.go

```diff
@@ -156,52 +172,40 @@ func (m *Monitor) Statistics(tags map[string]string) ([]*Statistic, error) {
 			statistic.Tags[k] = v
 		}
 
-		// Every other top-level expvar value is a map.
-		m := kv.Value.(*expvar.Map)
+		for k, v := range s.Tags() {
+			statistic.Tags[k] = v
+		}
 
-		m.Do(func(subKV expvar.KeyValue) {
-			switch subKV.Key {
-			case "name":
-				// straight to string name.
-				u, err := strconv.Unquote(subKV.Value.String())
-				if err != nil {
-					return
-				}
-				statistic.Name = u
-			case "tags":
-				// string-string tags map.
-				n := subKV.Value.(*expvar.Map)
-				n.Do(func(t expvar.KeyValue) {
-					u, err := strconv.Unquote(t.Value.String())
-					if err != nil {
-						return
-					}
-					statistic.Tags[t.Key] = u
-				})
-			case "values":
-				// string-interface map.
-				n := subKV.Value.(*expvar.Map)
-				n.Do(func(kv expvar.KeyValue) {
-					var f interface{}
-					var err error
-					switch v := kv.Value.(type) {
-					case *expvar.Float:
-						f, err = strconv.ParseFloat(v.String(), 64)
-						if err != nil {
-							return
-						}
-					case *expvar.Int:
-						f, err = strconv.ParseInt(v.String(), 10, 64)
-						if err != nil {
-							return
-						}
-					default:
-						return
-					}
-					statistic.Values[kv.Key] = f
-				})
-			}
-		})
+		if s.Refs() == 1 {
```

Comment:
Actually, there is a neat way to do this by introducing Owner and Observable interfaces which are supersets of a subset of the existing Statistics interface.

The Owner interface would have the Open, Close, Set*, Add* methods.

The Observable interface would have OpenObserver, CloseObserver, IsOwnerClosed() methods.